sidechain/consensus/slots/src/slots.rs

```diff
@@ -80,6 +80,14 @@ impl<ParentchainBlock: ParentchainBlockTrait> SlotInfo<ParentchainBlock> {
 			last_imported_parentchain_head: parentchain_head,
 		}
 	}
+
+	pub fn duration_remaining(&self) -> Option<Duration> {
+		let duration_now = duration_now();
+		if self.ends_at <= duration_now {
+			return None
+		}
+		Some(self.ends_at - duration_now)
+	}
```

Comment:
Added a convenience function to the `SlotInfo` struct to get the remaining duration in this slot.